route/router.go

```diff
@@ -39,6 +40,23 @@ type Router interface {
 	Plan() store.Variable[Plan]
 }
 
+// FleetPlan is an (incomplete) Plan that operates on the internal
+// solver data structures. Certain router options that customize solver
+// internals have to work with this data structure.
+type FleetPlan struct {
```

Comment:
I was first experimenting with that idea but settled with two different structs because the data structures used internally (ints mostly) and the data structures of a user facing result (Stops, Vehicles) are fundamentally different. In addition it is quite expensive at the moment to convert the internal data structures to an information richer format.